cg/meta/transfer/external_data.py

```diff
@@ -5,210 +5,204 @@
+        available_sample_ids: list[str] = [
+            sample.internal_id
+            for sample in self.status_db.get_samples_from_ticket(ticket=self.ticket)
             if sample.internal_id in available_folders or sample.name in available_folders
         ]
-        return available_samples
+        return available_sample_ids
 
-    def add_files_to_bundles(
-        self, fastq_paths: list[Path], last_version: Version, lims_sample_id: str
-    ):
-        """Adds the given fastq files to the the hk-bundle"""
-        for path in fastq_paths:
-            LOG.info(f"Adding path {path} to bundle {lims_sample_id} in housekeeper")
-            self.housekeeper_api.add_file(path=path, version_obj=last_version, tags=HK_FASTQ_TAGS)
-
-    def get_failed_fastq_paths(self, fastq_paths_to_add: list[Path]) -> list[Path]:
-        failed_sum_paths: list[Path] = []
-        for path in fastq_paths_to_add:
-            failed_path: Path | None = self.check_fastq_md5sum(path)
-            if failed_path:
-                failed_sum_paths.append(failed_path)
-        return failed_sum_paths
-
-    def get_fastq_paths_to_add(
-        self, customer: str, hk_version: Version, lims_sample_id: str
-    ) -> list[Path]:
-        paths: list[Path] = self.get_all_paths(lims_sample_id=lims_sample_id, customer=customer)
+    def _get_available_sample_ids(self) -> list[str]:
+        """Return a list of samples available for adding to Housekeeper."""
+        destination_folder_path: Path = self._get_destination_path()
+        LOG.debug(f"Checking available samples in {destination_folder_path}")
+        for sample_folder in destination_folder_path.iterdir():
+            self._curate_sample_folder(sample_folder=sample_folder)
+        available_sample_ids: list[str] = self._get_sample_ids_from_folder(destination_folder_path)
+        return available_sample_ids
+
+    def _get_fastq_paths_to_add(self, sample_id: str) -> list[Path]:
+        """Return the paths of fastq files that have not been added to the Housekeeper bundle."""
+        sample_folder: Path = self._get_destination_path(sample_id)
+        LOG.debug(f"Checking fastq files in {sample_folder}")
+        file_paths: list[Path] = [
+            sample_folder.joinpath(path)
+            for path in get_files_matching_pattern(directory=sample_folder, pattern="*.fastq.gz")
+        ]
+        LOG.debug(f"Found {len(file_paths)} fastq files for sample {sample_id}")
+        hk_version: Version = self.housekeeper_api.get_or_create_version(bundle_name=sample_id)
         fastq_paths_to_add: list[Path] = self.housekeeper_api.check_bundle_files(
-            file_paths=paths,
-            bundle_name=lims_sample_id,
+            file_paths=file_paths,
+            bundle_name=sample_id,
             last_version=hk_version,
             tags=HK_FASTQ_TAGS,
         )
+        LOG.debug(f"{len(fastq_paths_to_add)} samples are not yet in Housekeeper for {sample_id}")
```

Comment:
This can't be samples right? Or can you have "5 samples are not yet in housekeeper for <sample_id>"?

Reply:
Is it supposed to be "fastq files are not yet in housekeeper..."?